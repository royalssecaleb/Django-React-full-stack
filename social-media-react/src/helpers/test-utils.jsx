import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

function render(ui, { ...renderOptions } = {}) {
  const Wrapper = ({ children }) => <BrowserRouter>{children}</BrowserRouter>;

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react";
export { render };
