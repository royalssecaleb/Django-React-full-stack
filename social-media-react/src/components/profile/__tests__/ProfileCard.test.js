import { render, screen } from "../../../helpers/test-utils";
import TestRenderer from "react-test-renderer";
import ProfileCard from "../ProfileCard";
import { BrowserRouter } from "react-router-dom";

const userData = {
  id: "0590cd67-eacd-4299-8413-605bd547ea17",
  first_name: "Mossie",
  last_name: "Murphy",
  name: "Mossie Murphy",
  post_count: 3,
  email: "Mossie@yopmail.com",
  bio: "Omnis necessitatibus facere vel in est provident sunt tempora earum accusantium debitis vel est architecto minima quis sint et asperiores.",
  username: "MossieMurphy",
  avatar: null,
  created: "2022-08-19T17:31:03.310Z",
  updated: "2022-08-20T07:38:47.631Z",
};

test("Render ProfileCard component", () => {
  render(<ProfileCard user={userData} />);

  const profileCard = screen.getByTestId("profile-card");
  expect(profileCard).toBeInTheDocument();

  const nameElement = screen.getByText(userData.name);
  expect(nameElement).toBeInTheDocument();
});

test("Profile Card snapshot", () => {
  const profileCardDomTree = TestRenderer.create(
    <BrowserRouter>
      <ProfileCard user={userData} />
    </BrowserRouter>
  ).toJSON();
  expect(profileCardDomTree).toMatchSnapshot();
});
