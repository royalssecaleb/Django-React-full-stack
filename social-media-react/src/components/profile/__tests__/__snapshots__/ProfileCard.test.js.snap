// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Card snapshot 1`] = `
<div
  className="border-0 p-2 card"
  data-testid="profile-card"
>
  <div
    className="d-flex "
  >
    <img
      className="my-3 border border-primary border-2 rounded-circle"
      height={48}
      src={null}
      width={48}
    />
    <div
      className="card-body"
    >
      <div
        className="fs-6 card-title h5"
      >
        Mossie Murphy
      </div>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        See profile
      </button>
    </div>
  </div>
</div>
`;
